Documentation changes

Jared-Krajewski opened the pull request:
overhaul to documentation of readme, contributing, resource, roadmap as well as bug and feature templates

Jared-Krajewski committed:
changes to content and formatting of README.MD

Jared-Krajewski committed:
fixed formatting in resources section

Jared-Krajewski committed:
added line break between sections 4 and 5

Jared-Krajewski committed:
added line break between sections 4 and 5

Jared-Krajewski committed:
added back to top links in overview and tech stack sections

Jared-Krajewski committed:
added setup instructions and tech stack image

Jared-Krajewski committed:
added setup instructions and tech stack image

Jared-Krajewski committed:
added return to top to setup and removed bullet formatting from tech stack img

Jared-Krajewski committed:
added several new .md files to docs with content changes for all documentation

Jared-Krajewski committed:
formatting and content changes to all documentation files

Jared-Krajewski committed:
layout and content changes to contributing.md and minor formatting and content changes to other documentation

Jared-Krajewski committed:
content changes to development process of CONTRIBUTING.md

Jared-Krajewski committed:
removed html from readme.md and minor content changes

Jared-Krajewski committed:
formatting to docs/readme and resources to increase readability

Jared-Krajewski committed:
formatting and content changes to readme and contributing

Jared-Krajewski committed:
additional organizational and formatting changes to readme and contributing

leekahung commented on docs/RESOURCES.MD:
Since the script from `clearDocs.js` looks for `.md` files to ignore when removing files in `/docs`, it will delete the `.MD` files if someone ran `npm run docs:clear`. Think it'll be better to change the extension here to `.md` rather than adding additional conditions to `clearDocs.js`

Jared-Krajewski replied:
The extension is lowercase locally but doesn't change when I push. I will try manually in GitHub

leekahung commented on docs/ROADMAP.MD:
Same comment as above. Think it'll be better to change the extension from `.MD` to `.md` to avoid the markdown file from being deleted from `npm run docs:clear`. See clip.

https://github.com/codeforpdx/PASS/assets/14917816/2dcbe457-54df-45a9-b6ba-861e9fee6234

Jared-Krajewski committed:
added placeholder logo to readme and more formatting to readme and contributing

Jared-Krajewski committed:
Rename RESOURCES.MD to RESOURCES.md

changed file type from MD to md directly in github since it would not change on remote push.

Jared-Krajewski committed:
Rename ROADMAP.MD to ROADMAP.md

changed file type from MD to md directly in GitHub since remote push would not change it.

leekahung commented on docs/CONTRIBUTING.md:
```diff
@@ -0,0 +1,138 @@
+
+- Good people to ping for reviews:
+  - Development -- Jared K, Ka Hung L., Kevin M., Tim S., Scott B.
+  - Documentation -- Danica, Florian(flo), Jared K
```
Think we can change this to: `Documentation -- Danica, Florian (Flo), Jared K`

Jared-Krajewski replied:
done

leekahung commented on docs/RESOURCES.md:
```diff
@@ -0,0 +1,50 @@
+<a href="https://www.inrupt.com/"><img src="https://docs.inrupt.com/inrupt-logo-small.svg" alt="InruptIcon" width="18" height="18"> Inrupt(solid library)
+<a href="https://eslint.org/"><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/e/e3/ESLint_logo.svg/324px-ESLint_logo.svg.png?20211012234406" alt="SolidProjectIcon" width="18" height="18"> ESLint
+🗒️  [JSDoc](https://jsdoc.app/)
+  ![tech stack](https://drive.google.com/uc?id=1hvs0XKI0hfdlGOJ_oRG-MqQTR9NCtmPy)
```
Think we can split this into two sections, one for the shields, one for the image itself. Otherwise, it'll stretch out the entire section.

https://github.com/codeforpdx/PASS/assets/14917816/5dfff484-770b-48ed-aefd-83d34782fc21

Jared-Krajewski replied:
added it to a section

Jared-Krajewski committed:
added badges in resources to section to prevent bleeding over, spelling change to contributing for documentation

leekahung commented on docs/CONTRIBUTING.md:
```diff
@@ -0,0 +1,138 @@
+   - `git clone https://github.com/codeforpdx/PASS.git` [learn more about git clone](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository?platform=linux)
+   - `git remote add origin https://github.com/codeforpdx/PASS.git` [learn more about git remote](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories)
+
+2. Setup instructions to locally run PASS can be found in the [readme](../README.md)
```
I believe the link here refers to the more detailed instructions for running PASS locally on `/docs/README.md` rather than the short version on `README.md`?

Jared-Krajewski replied:
/docs/README.md has the instructions as well but they are sort of hidden amongst many other. Maybe making the two uniform or removing them from /docs/README and keeping it in the main README? Tim had made a previous comment about the importance of having those instructions in the main README and it seems to be common practice

leekahung replied:
The `README.md` in `/docs` are there to be used as the main page for `JSDoc`. Think we should keep what we have there.

We could duplicate what we have there for the instructions onto the main `README.md`

Jared-Krajewski replied:
resolved in discord VC, linked in main README

leekahung commented on docs/RESOURCES.md:
```diff
@@ -0,0 +1,53 @@
+[![Material UI](https://img.shields.io/badge/Material%20UI-007FFF?style=for-the-badge&logo=mui&logoColor=white)](https://mui.com/)
+<a href="https://solidproject.org"><img src="https://solidproject.org/assets/img/solid-emblem.svg" alt="SolidProjectIcon" width="18" height="18"> Solid Project
+<a href="https://www.inrupt.com/"><img src="https://docs.inrupt.com/inrupt-logo-small.svg" alt="InruptIcon" width="18" height="18"> Inrupt(solid library)
+<a href="https://eslint.org/"><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/e/e3/ESLint_logo.svg/324px-ESLint_logo.svg.png?20211012234406" alt="SolidProjectIcon" width="18" height="18"> ESLint
```
`ESLint` appeared twice. I believe one of them is supposed to be `Prettier` since it's not included here?

Jared-Krajewski replied:
removed second ESLint and added prettier, will be in next push

timbot1789 commented on README.md:
```diff
@@ -1,58 +1,127 @@
+
+- ### Prerequisites
+
+   Currently we require Node version 16 or higher and NPM for our package manager.  Most places recommend using a node version manager to install node and npm. To proceed using NVM perform the following.. 
```
Thanks for adding this. Looks good!

leekahung replied:
Yeah, I agree with Tim. The new updates for our documentation are coming out nicely.

timbot1789 commented on docs/CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,39 @@
+# Code of Conduct 
+(impending updates on the horizon)
+
+We believe that anyone attending a Code for America event or participating in our online community should feel safe and be free from harassment.
```
I was under the impression that we are no longer part of code for america. Maybe pass this by Hugh and see what the best thing to call ourselves is?

Jared-Krajewski replied:
duplicate comment? leaving the other one open because it has two posts.

timbot1789 commented on docs/CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,39 @@
+# Code of Conduct 
+(impending updates on the horizon)
+
+We believe that anyone attending a Code for America event or participating in our online community should feel safe and be free from harassment.
```
I was under the impression that we are no longer part of code for america. Maybe pass this by Hugh and see what the best thing to call ourselves is?

timbot1789 replied:
I think we should be fine using the code of conduct itself, but just replace 'Code for America' with whatever our name is.

Jared-Krajewski replied:
wasn't sure if it was transitional or not still. I will follow up and edit/update as appropriate but this is what was available and just copy/pasted from the current code for pdx website.

Jared-Krajewski replied:
I asked Hugh, he said to just link to the code of conduct that is pinned to the code for pdx github (which this was just copy and pasted from) so the reference is fine until it is actually updated.

timbot1789 commented on docs/CONTRIBUTING.md:
```diff
@@ -0,0 +1,138 @@
+
+2. Setup instructions to locally run PASS can be found in the [readme](../README.md)
+
+3. Create a new branch to work on your feature:
```
You can also create a branch directly from an issue:
![image](https://github.com/codeforpdx/PASS/assets/37914436/4533d67e-e398-4561-8b90-bdbffbca9a6c)

Jared-Krajewski replied:
I was thinking about adding cropped screenshots to all of the dev instructions that involve a GUI, not sure if you think its necessary or not. I will add this step with the next push.

timbot1789 commented on docs/CONTRIBUTING.md:
```diff
@@ -0,0 +1,138 @@
+   If you are new to GitHub and/or the team, feel free to make your first pull request on the README/Contributing documentation to familiarize yourself with the project and GitHub. Add any comments and/or feedback and request reviews.
+1. Make a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to the `Development` branch. Request reviews from members of the team - you’ll need their approval to merge. \*\*Make sure to close your branch once merged.
+- ## Bug Reporting Template
+  Bugs are reported via the git-hub built in issues for the repo https://github.com/codeforpdx/PASS/issues. 
```
typo: github built-in

Jared-Krajewski replied:
fixed and will be in next push

Jared-Krajewski committed:
added line to contributing for branch directly from issue and edited recomnended reviewers

Jared-Krajewski committed:
added links to bug reporting and feature request as well as formatting changes to contributing

Jared-Krajewski committed:
removed code of conduct and referenced Code for PDX code of conduct for rebranding changes

Jared-Krajewski committed:
removed extra char from contributors and fixed back to top pathing

Jared-Krajewski committed:
updated templates for bug report and feature request, re worded parts of contributing

Jared-Krajewski committed:
removing emoji from setup instructions to test linking contents

Jared-Krajewski committed:
remove emojis to fix TOC linking

Jared-Krajewski committed:
updated tech stack img, removed dead and redundant links, removed solid UI ref as we are soon to not be using, removed linting steps and referenced in link to readme

Jared-Krajewski committed:
added enhancement request and description with linking in contributing

Jared-Krajewski committed:
added quotes on example for unit testing

Jared-Krajewski committed:
updated branch naming convention

Jared-Krajewski committed:
Merge branch 'Development' into Documentation_changes

Jared-Krajewski committed:
spelling and formatting changes

Jared-Krajewski committed:
moved logo to assets and fixed linking

Jared-Krajewski committed:
changed MD to md :D

Jared-Krajewski committed:
added link for digita in resources

Jared-Krajewski committed:
updated discord invite link

Jared-Krajewski committed:
fixed type in contributing

leekahung reviewed (approved):
Have commented and gone through the changes together on voice chat. I'm good with what we have here, but is up for future changes/additions.

I'll approve this.

Jared-Krajewski committed:
Merge branch 'Development' into Documentation_changes
